src/goats_tom/tasks/check_latest_version.py

```diff
@@ -0,0 +1,135 @@
+"""Check and publish the latest GOATS version info (WS + Redis cache)."""
+
+__all__ = ["check_version_info"]
+
+import logging
+from importlib.metadata import version as get_version
+from json import JSONDecodeError
+
+import dramatiq
+import requests
+from asgiref.sync import async_to_sync
+from channels.layers import get_channel_layer
+from django.core.cache import caches
+from packaging.version import InvalidVersion, Version
+
+logger = logging.getLogger(__name__)
+
+PACKAGE_NAME = "goats"
+CHANNELDATA_URL = "https://gemini-hlsw.github.io/goats-infra/conda/channeldata.json"
+GROUP_NAME = "updates_group"
+CACHE_KEY = "version:info"
+
+
+def _notify_version_info_ws(is_outdated: bool, current: str, latest: str) -> None:
+    """Push a version snapshot to the ``updates_group`` WebSocket group.
+
+    Sends a Django Channels group message with ``type="version.info"`` that
+    includes whether the installed version is outdated and the current/latest
+    version strings.
+
+    Parameters
+    ----------
+    is_outdated : bool
+        ``True`` if the installed version is older than the latest available.
+    current : str
+        Installed package version string.
+    latest : str
+        Latest available package version string resolved from the channel.
+    """
+    try:
+        channel_layer = get_channel_layer()
```

Comment:
I had written a helper class that does all of this called `goats_tom.realtime.NotificationInstance.create_and_send`. I use that in my file. My bad for not telling you.